thingsboard_gateway/storage/sqlite/sqlite_event_storage.py

```diff
@@ -1,113 +1,352 @@
-#     Copyright 2025. ThingsBoard
-#
-#     Licensed under the Apache License, Version 2.0 (the "License");
-#     you may not use this file except in compliance with the License.
-#     You may obtain a copy of the License at
-#
-#         http://www.apache.org/licenses/LICENSE-2.0
-#
-#     Unless required by applicable law or agreed to in writing, software
-#     distributed under the License is distributed on an "AS IS" BASIS,
-#     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-#     See the License for the specific language governing permissions and
-#     limitations under the License.
-
+import copy
 from gc import collect
-from sqlite3 import InterfaceError
+from logging import getLogger
+from os import path, makedirs, remove
+from queue import Queue, Full
+from sqlite3 import ProgrammingError, DatabaseError
 from threading import Event
-from time import sleep, time, monotonic
+from time import sleep, monotonic
 
 from thingsboard_gateway.storage.event_storage import EventStorage
 from thingsboard_gateway.storage.sqlite.database import Database
-from queue import Queue, Full
-from logging import getLogger
+from thingsboard_gateway.storage.sqlite.sqlite_event_storage_pointer import Pointer
+from thingsboard_gateway.storage.sqlite.storage_settings import StorageSettings
 
 
 class SQLiteEventStorage(EventStorage):
-    """
-    HIGH level api for thingsboard_gateway main loop
-    """
 
     def __init__(self, config, logger, main_stop_event):
         super().__init__(config, logger, main_stop_event)
         self.__log = logger
         self.__log.info("Sqlite Storage initializing...")
         self.write_queue = Queue(-1)
         self.stopped = Event()
-        self.db = Database(config, self.write_queue, self.__log, stopped=self.stopped)
-        self.db.init_table()
-        self.__log.info("Sqlite storage initialized!")
-        self.delete_time_point = None
+        self.__config_copy = copy.deepcopy(config)
```

Comment:
It will be better to work with storage settings object, instead of copying dict with config. You can adjust storage settings, for example make copy() method, which will return a new instance with the same values in attributes. And then adjust a new settings object with required data.